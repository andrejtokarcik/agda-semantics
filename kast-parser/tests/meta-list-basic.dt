data List (a : Set 0) : Set 0 where
  Nil : List a,
  Cons : Pi x a (Pi y (List a) (List a))
;
data Nat : Set 0 where
  Zero : Nat,
  Succ : Pi x Nat Nat

; expr = Cons ? (Succ Zero) (Nil ?)
