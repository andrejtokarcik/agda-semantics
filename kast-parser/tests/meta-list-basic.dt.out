<k>
    nf2 ( (  Cons #symExpr(10)  Succ Zero   Nil #symExpr(21)   ,  List Nat
        ) )
</k>
<mgu>
    subst(.KList)
</mgu>
<tenv>
    .Map
</tenv>
<data>
    Cons |-> (Pi a Set 0 (Pi x a (Pi y List a List a)))
    List |-> (Pi a Set 0 Set 0)
    Nat |-> Set 0
    Nil |-> (Pi a Set 0 List a)
    Succ |-> (Pi x Nat Nat)
    Zero |-> Nat
</data>
<meta>
    #symExpr(10) |-> ( Nat , Set 0 )
    #symExpr(21) |-> ( Nat , Set 0 )
</meta>
